src/main/java/Duke.java

```diff
@@ -1,10 +1,141 @@
+import java.util.Scanner;
+import java.util.Vector;
+
 public class Duke {
+    public class Task {
+        protected String description;
+        protected boolean isDone;
+        protected String taskType;
+        protected String taskDate;
+    
+        public Task(String description,boolean isDone, String taskType, String taskDate) { // initiating an instance
+            this.description = description;
+            this.isDone = isDone;
+            this.taskType = taskType; // todo,ddl,event
+            this.taskDate = taskDate; // 
+
+        }
+
+        public String getTaskType(){
+            return (this.taskType);
+        }
+    
+        public String getStatusIcon() {
+            return (isDone ? "X" : " "); // mark done task with X
+        }
+    
+        public void markAsDone(){
+            this.isDone = true;
+        }
+
+        public void markAsNotDone(){
+            this.isDone = false;
+        }
+
+        public String getDescription(){
+            return this.description;
+
+        }
+
+        public String getDate(){
+            return this.taskDate;
+    }
+}
```

Comment:
Maybe you can consider having a space after the closing bracket for the method instead?
eg.
public String getDate() {
//Insert code here
}